projects/RabbitMQ.Client/client/impl/Connection.cs

```diff
@@ -395,6 +397,29 @@ internal void OnCallbackException(CallbackExceptionEventArgs args)
 
         internal void Write(ReadOnlyMemory<byte> memory)
         {
+            Activity? activity = Activity.Current;
+            if (activity != null && activity.IsAllDataRequested)
```

Comment:
@bollhals The null and `.IsAllDataRequested` checks should prevent most allocations from taking place when tracing is disabled.